### ISSUE TITLE
Floating-point literals should store bits in integers, not use host f32/f64.

### ISSUE THREAD

**Issue by eddyb** [gfx-rs/rspirv#189]
Compilers should never rely on host floating-point support, as that can lead to host-specific behavioral subtleties (instead of behavior being deterministically tied to the target architecture/platform), and in many cases there's not enough to gain to even look into what guarantees certain host architectures/platforms may offer.

For `rustc` we ported LLVM's C++ `APFloat` library to Rust (for e.g. evaluating float operations in constant expressions), but `rspirv` itself only needs to store literals, not operate on the actual values (and consumers could choose how to interpret the bits of a floating-point value, any way they want).

I'm not sure how what the timeline for 0.8 is, but this would be a **breaking change**.

Also, it would help with #16.

cc @khyperia

**Comment by Jasper-Bekkers** [comment]
@eddyb do you have any examples of where this is done wrong? Is it just the `constant_f32` functions and `LiteralFloat32` etc?

**Comment by eddyb** [comment]
Right, and loading code like this: https://github.com/gfx-rs/rspirv/blob/719cf08e4af0436242707479e3509add5ec3d514/rspirv/binary/decoder.rs#L205-L219

Basically the types `f32` and `f64` should more or less not show up in `rspirv` (except maybe for `fmt::Debug`?).
